### PR TITLE
Group tables by DATA_DESC_ID when present

### PR DESCRIPTION
**Pull request by astrofrog**
This also cleans up the API to use the standard astropy Table reader mechanism. With this, one can do e.g.:

```python
    >>> import casa_formats_io
    >>> from astropy.table import Table
    >>> table = Table.read('my_dataset.ms', format='casa-table')
```

I used the following script to check a 10Gb MS file and verified that all the data match between casatools and casa-formats-io:

```python
import numpy as np
from casatools import table
import dask.array as da
import casa_formats_io
from astropy.table import Table
from numpy.testing import assert_equal

table_filename = 'W51-IRS2_B3_uid___A001_X1296_X18f_continuum_merged_12M_selfcal.ms'

data_desc = Table.read(table_filename + '/DATA_DESCRIPTION', format='casa-table')


tb = table()
tb.open(table_filename)

for data_desc_id in range(len(data_desc)):

    print(f'DATA_DESC_ID={data_desc_id}')

    table = Table.read(table_filename, format='casa-table', data_desc_id=data_desc_id)

    sub_tb = tb.query(f'DATA_DESC_ID=={data_desc_id}')

    for colname in sub_tb.colnames():

        if colname == 'FLAG_CATEGORY':
            continue

        reference = sub_tb.getcol(colname).T
        actual = table[colname]
        if isinstance(actual, da.Array):
            actual = actual.compute()

        if reference.size == 0 and actual.size == 0:
            continue

        assert_equal(actual, reference)

        print(f' - {colname} matches')
```

There is a test failure I need to track down before we can merge, but this works fine with real-world data it seems!

### CODE REVIEW

**Comment by astrofrog** [comment]
The test failure is due to https://github.com/dask/dask/issues/8387, waiting for input on that issue before proceeding.

**Comment by astrofrog** [comment]
I think this is ready for review/merging (review-wise, the main bit that is important is probably the docs)

File: docs/index.rst [hunk 1]
```diff
@@ -44,6 +44,23 @@ respectively.
...
+---------------------------
+
+This package includes an experimental generic table reader which integrates with
+the astropy :class:`~astropy.table.Table` class. To use it, first import
+the ``casa_formats_io`` module, which registers the reader, then use the
+:meth:`Table.read <astropy.table.Table.read>` method::
+
+    >>> import casa_formats_io
+    >>> from astropy.table import Table
+    >>> table = Table.read('my_dataset.ms', format='casa-table')
```
Comment (keflavich) [inline docs/index.rst:57]: could one do:
```python
from casa_formats_io import Table
table = Table.read('my_dataset.ms', format='casa-table')
```
and still have `Table` be the astropy default table?  I think this small bit of convenience wrapping would be nice.  I wouldn't suggest changing the docs for this, but perhaps add it.

File: docs/index.rst [hunk 2]
```diff
@@ -44,6 +44,23 @@ respectively.
...
+
+If the table contains a ``DATA_DESC_ID`` column, which is the case for e.g.
+measurement sets, you will need to also specify the ``data_desc_id=`` argument
+to :meth:`Table.read <astropy.table.Table.read>` with a valid integer
+DATA_DESC_ID value.
```
Comment (keflavich) [inline docs/index.rst:62]:
```suggestion
DATA_DESC_ID value.

    >>> table_3 = Table.read('my_multims.ms', format='casa-table', data_desc_id=3)
```

**Review by keflavich — APPROVED** [review]
minor suggestions on docs.  Looks good!

**Comment by astrofrog** [comment]
I'll go ahead and merged since approved!

**Pull request merged**